test/test_puma_localhost_authority.rb

```diff
@@ -0,0 +1,168 @@
+# Nothing in this file runs if Puma isn't compiled with ssl support
+#
+# helper is required first since it loads Puma, which needs to be
+# loaded so HAS_SSL is defined
+require_relative "helper"
+
+if ::Puma::HAS_SSL  && !Puma.jruby?
+  require "puma/events"
+  require "net/http"
+  require "localhost/authority"
+
+  class SSLEventsHelper < ::Puma::Events
+    attr_accessor :addr, :cert, :error
+
+    def ssl_error(error, ssl_socket)
+      self.error = error
+      self.addr = ssl_socket.peeraddr.last rescue "<unknown>"
+      self.cert = ssl_socket.peercert
+    end
+  end
+
+  # net/http (loaded in helper) does not necessarily load OpenSSL
+  require "openssl" unless Object.const_defined? :OpenSSL
+
+  puts "", RUBY_DESCRIPTION, "RUBYOPT: #{ENV['RUBYOPT']}",
+    "                         OpenSSL",
+    "OPENSSL_LIBRARY_VERSION: #{OpenSSL::OPENSSL_LIBRARY_VERSION}",
+    "        OPENSSL_VERSION: #{OpenSSL::OPENSSL_VERSION}", ""
+end
+
+class TestPumaLocalhostAuthority < Minitest::Test
```

Comment:
Are these tests copied from another file?

I think for this feature we should be testing:

1. If the binder is SSL but there is no SSL cert provided, a self-signed one is generated (and used)
2. The self-signed cert is signed by the local authority
3. That's probably it. We know SSL works in Puma from other tests.